### PR TITLE
Refactor/media settings modal

### PR DESCRIPTION
**Pull request by alexanderleegs**
Previously, we were unable to simply refetch the data as `MediaSettingsModal` was still a class component. This PR refactors the `MediaSettingsModal` into a functional component, and adds react-query to it, in order to handle updates to media details by refetching data instead of reloading the page. 

In addition, this PR also changes the type parameter of MediaSettingsModal to expect `images` and `files` instead of `image` and `file`, for greater consistency with the rest of the components.

### CODE REVIEW

File: src/components/media/MediaSettingsModal.jsx [hunk 1]
```diff
@@ -10,221 +10,184 @@ import { validateFileName } from '../../utils/validators';
...
+export default MediaSettingsModal
+
 MediaSettingsModal.propTypes = {
   media: PropTypes.shape({
     fileName: PropTypes.string,
     path: PropTypes.string,
     content: PropTypes.string,
   }).isRequired,
-  type: PropTypes.oneOf(['image', 'file']).isRequired,
+  type: PropTypes.oneOf(['images', 'files']).isRequired,
```
Comment (gweiying) [inline src/components/media/MediaSettingsModal.jsx:190]: nit: missing `customPath` as propType

File: src/components/media/MediaSettingsModal.jsx [hunk 2]
```diff
@@ -10,221 +10,184 @@ import { validateFileName } from '../../utils/validators';
...
+    }
+  )
+
+  // Handling delete
+  const { mutateAsync: deleteHandler } = useMutation(
+    () => deleteMedia({siteName, type, sha, customPath, fileName}),
+    {
+      onError: () => errorToast(`There was a problem trying to delete this ${type === 'images' ? 'image' : 'file'}. ${DEFAULT_RETRY_MSG}`),
+      onSuccess: () => {
+        successToast(`Successfully deleted ${type.slice(0,-1)}!`)
```
Comment (gweiying) [inline src/components/media/MediaSettingsModal.jsx:85]: nit: Let's standardize on `${type === 'images' ? 'image' : 'file'}` which seems more readable?

File: src/components/media/MediaSettingsModal.jsx [hunk 3]
```diff
@@ -10,221 +10,184 @@ import { validateFileName } from '../../utils/validators';
...
+        } else if (err?.response?.status === 413 || err?.response === undefined) {
+          // Error due to file size too large - we receive 413 if nginx accepts the payload but it is blocked by our express settings, and undefined if it is blocked by nginx
+          errorToast(`Unable to upload as the ${type === 'images' ? 'image' : 'file'} size exceeds 5MB. Please reduce your ${type === 'images' ? 'image' : 'file'} size and try again.`)
+        } else {
+          errorToast(`There was a problem trying to save this ${type === 'images' ? 'image' : 'file'}. ${DEFAULT_RETRY_MSG}`)
+        }
+        console.log(err);
+      },
+      onSuccess: () => {
+        successToast(`Successfully ${isPendingUpload ? `created new` : `renamed`} ${type.slice(0,-1)}!`)
```
Comment (gweiying) [inline src/components/media/MediaSettingsModal.jsx:70]: nit: Let's standardize on ${type === 'images' ? 'image' : 'file'} which seems more readable?

**Review by gweiying — APPROVED** [review]
overall looks great, just some small nits as below!

File: src/components/media/MediaSettingsModal.jsx (hunk 1 above)
Comment (alexanderleegs) [reply src/components/media/MediaSettingsModal.jsx:190]: Good catch, resolved in https://github.com/isomerpages/isomercms-frontend/pull/453/commits/f07d4c3f03fd5db543ea8fbc1b5911f4573be45e

File: src/components/media/MediaSettingsModal.jsx [hunk 4]
```diff
@@ -10,221 +10,184 @@ import { validateFileName } from '../../utils/validators';
...
-                isRequired
-                onFieldChange={this.setFileName}
+  }, [mediaData])
+
+  return (
+    <div className={elementStyles.overlay}>
+      <div className={elementStyles.modal}>
+        <div className={elementStyles.modalHeader}>
+          <h1>
+            {isPendingUpload ? `Upload new ${type}` : `Edit ${type} details`}
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/components/media/MediaSettingsModal.jsx]: https://github.com/isomerpages/isomercms-frontend/pull/453/commits/f84697a9fe95cc27d563d5a80769d6d42920a0ab

File: src/components/media/MediaSettingsModal.jsx (hunk 3 above)
Comment (alexanderleegs) [reply src/components/media/MediaSettingsModal.jsx:70]: https://github.com/isomerpages/isomercms-frontend/pull/453/commits/f84697a9fe95cc27d563d5a80769d6d42920a0ab

**Review by gweiying — APPROVED** [review]
lgtm

**Pull request merged**